packages/x-date-pickers/src/DateField/DateField.tsx

```diff
@@ -79,10 +97,78 @@ DateField.propTypes = {
    * @default false
    */
   disablePast: PropTypes.bool,
+  /**
+   * If `true`, the label is displayed in an error state.
+   * @default false
+   */
+  error: PropTypes.bool,
```

Comment:
To remove, we are always overriding it
If we want people to be able to override the `error` attribute, we should edit the behavior to put it before the spread (could be discussed)